database.hh

```diff
@@ -1250,7 +1250,7 @@ struct string_pair_eq {
 //   local metadata reads
 //   use shard_of() for data
 
-class database {
+class database : public peering_sharded_service<database> {
```

Comment:
Why not initialize those virtual tables with sharded<database>? If they're node-wide virtual tables, it should be explicit.

Reply:
Good idea, although this requires the refactoring of the virtual table registering mechanism to run on shard 0 only and take `sharded<VirtualTable>` instances. For this we'd also need `sharded<>` to work with polimorphic objects: we want to register `sharded<virtual_table>` instances so we can store them in a homogeneous container.

Reply:
I don't understand. We can register the virtual table on each shard, but give it access to sharded<database>.